util/plaidbench/get-optsched-stats.py

```diff
@@ -1,242 +1,358 @@
-        print("      Regions passed to B&B: {} ({:.1f}%)".format(secondPassEnumCnt, float(secondPassEnumCnt)/secondPassTotalProcessed*100.0))
-        print("      Regions optimal and improved: {} ({:.1f}%)".format(secondPassOptImpr, float(secondPassOptImpr)/secondPassEnumCnt*100.0))
-        print("      Regions optimal and not improved: {} ({:.1f}%)".format(secondPassOptNotImpr, float(secondPassOptNotImpr)/secondPassEnumCnt*100.0))
-        print("      Regions timed out and improved: {} ({:.1f}%)".format(secondPassTimeoutImpr, float(secondPassTimeoutImpr)/secondPassEnumCnt*100.0))
-        print("      Regions timed out and not improved: {} ({:.1f}%)".format(secondPassTimeoutNotImpr, float(secondPassTimeoutNotImpr)/secondPassEnumCnt*100.0))
-        print("      Avg. region size passed to B&B: {:.1f}".format(secondPassTotalInstr/secondPassEnumCnt))
-        print("      Largest optimal region: {}".format(secondPassLargestOptimalRegion))
-        print("      Largest improved region: {}".format(secondPassLargestImprovedRegion))
-    else:
-        print("      Second pass did not pass any regions to B&B enumerator")
+            print('Cannot find log file for benchmark {}.'.format(bench))
+
+        for passNum in passes:
+            for stat in statsProcessed:
+                passStats[passNum][stat] += stats[passNum][stat]
+            if stats[passNum]['EnumCnt'] != 0:
+                stats[passNum]['AverageSizeToEnum'] = float(stats[passNum]['TotalInstr'])/stats[passNum]['EnumCnt']
+            if passStats[passNum]['LargestOptimalRegion'] < stats[passNum]['LargestOptimalRegion']:
+                passStats[passNum]['LargestOptimalRegion'] = stats[passNum]['LargestOptimalRegion']
+            if passStats[passNum]['LargestImprovedRegion'] < stats[passNum]['LargestImprovedRegion']:
+                passStats[passNum]['LargestImprovedRegion'] = stats[passNum]['LargestImprovedRegion']
+
+        benchStats[bench] = stats
+
+    for passNum in passes:
+        if passStats[passNum]['EnumCnt'] != 0:
+            passStats[passNum]['AverageSizeToEnum'] = float(passStats[passNum]['TotalInstr'])/passStats[passNum]['EnumCnt']
+
+def printStats():
+    for passNum in passes:
+        if passStats[passNum]['TotalProcessed'] == 0:
+            continue
+
+        # Third pass is just a default if the two-pass
+        # implementation wasn't used and not an actual
+        # 3 pass implementation.
+        if passNum == 'third':
+            print('first')
+        else:
+            print(passNum)
+    
+        for stat in passStats[passNum]:
+            print('    {} : {}'.format(stat, passStats[passNum][stat]))
+
+def writeBenchmarkNames(ws, row):
+    for bench in benchmarks:
+        ws['A' + str(row)] = bench
+        row += 1
+    ws['A' + str(row)] = 'Overall'
+    ws['A' + str(row)].font = Font(bold=True)
+
+def createSpreadsheets(output):
+    if 'xls' not in output[-4:]:
```

Comment:
should this be xls or xlsx?

Reply:
It should be xls but it can also detect xlsx. Both are valid formats for excel spreadsheets. xls is an older format and I only kept xls because the xlwt module used in 'extract-plaidbench-data.py' saves spreadsheets in xls format. I just recently found out that xlwt is no longer being updated and can swap to openpyxl, which uses the latest format, if this is an issue.

Reply:
Ah, ok. Got it. Was just wondering about it because the line below says xlsx.